### PR TITLE
Distance request

### LINKED ISSUE #6: Add location service to Weather Micro-Service.

### LINKED ISSUE #9: End to End Connection
- [x] BE engine is talking to microservice and fully functioning

### PR DESCRIPTION
**Pull request by jmichlitsch**
### What Changed?
- Add Happy Path for Distance Controller Endpoint.
### What's Next?
- Add Sad Paths for Controller
### Known Issues?
- Just need to add sad paths and get sinatra errors working properly
### Test Coverage?
- 100%
### README & Documentation
- _Think: Does this PR add something that should be documented? If not, you can remove this section._
- What did you add to the README or project documentation?

### CODE REVIEW

File: spec/requests/distance_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require './app/services/map_quest_service'
+require 'rspec'
+require 'rack/test'
+require 'spec_helper'
+require './app/controllers/map_quest_service_app'
+
+
+describe 'Climate Data API' do
```
Comment (klatour324) [inline spec/requests/distance_request_spec.rb:8]: I think this is supposed to be Distance here.

File: spec/requests/distance_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      parsed = JSON.parse(last_response.body, symbolize_names: true)
+      data = parsed[:data]
+
+      expect(data).to be_a(Hash)
+      # expect(data[:type]).to eq("climate")
+      # expect(data[:type]).to be_a(String)
+      # expect(data[:id]).to eq(nil)
+      # expect(data.keys).to eq([:id, :type, :attributes])
+      # expect(data[:attributes]).to be_a(Hash)
+      # expect(data[:attributes].keys).to eq([:temp, :precip, :vintage, :region, :start_date, :end_date])
```
Comment (klatour324) [inline spec/requests/distance_request_spec.rb]: Don't forge to remove these commented out lines.

**Review by klatour324 — COMMENTED** [review]
Looks good Jeremiah. Just a couple of cleanups here otherwise great work!

**Review by klatour324 — APPROVED** [review]
Thanks for making those changes!

File: app/controllers/distance_service_app.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+require './app/facades/distance_facade'
+require 'sinatra'
+require 'json'
+require 'fast_jsonapi'
+require 'sinatra/json'
+
+class DistanceServiceApp < Sinatra::Base
+  #get '/' do
+  #   "Hello, World!"
+  #end
```
Comment (amsmyth1) [inline app/controllers/distance_service_app.rb:13]: Did you mean to remove this commented out code? or is this left here intentionally?

**Review by amsmyth1 — APPROVED** [review]
Looks good to me! I just had one note about some commented out code in the distance_service_app (lines 11-13). So cool to see a Poro working in Sinatra!

File: app/controllers/distance_service_app.rb (hunk 3 above)
Comment (jmichlitsch) [reply app/controllers/distance_service_app.rb:13]: I did. Katy added a root which would specify that path, it was also just to confirm I was at 100% test coverage with the methods I created.

**Pull request merged**